### PR TITLE
[e2e]: shrink utils.go with regards to storage and libpod

### PR DESCRIPTION
**Pull request by enp0s3**
Signed-off-by: Igor Bezukh <ibezukh@redhat.com>

### What this PR does
storage related functions copyAlpineWithNonQEMUPermissions deleteAlpineWithNonQEMUPermissions and createExecutorPodWithPVC were moved to storage package and became private.

RunPod* related helpers moved to libpod, tests were refactored to use the libpod where needed.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note

```

### CODE REVIEW

File: tests/framework/storage/nfs.go [hunk 1]
```diff
@@ -3,22 +3,23 @@ package storage
 import (
 	"fmt"
 
+	"kubevirt.io/kubevirt/tests/libpod"
```
Comment (ormergi) [inline tests/framework/storage/nfs.go:6]: nit: I would move it closer to tests dependencies group (line16)

File: tests/storage/memorydump.go [hunk 2]
```diff
@@ -28,6 +28,9 @@ import (
 	"strings"
 	"time"
 
+	"kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libpod"
```
Comment (ormergi) [inline tests/storage/memorydump.go:32]: nit: Move to test dependencies group (line 55), although test imports is all over the place.

File: tests/libpod/run.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package libpod
+
+import (
+	"context"
+
+	. "github.com/onsi/gomega"
```
Comment (ormergi) [inline tests/libpod/run.go:25]: nit: add new line for tests framework dependencies group

**Review by ormergi — APPROVED** [review]
Looks good

I left some nits regarding the import ordering.

**Review by dankenigsberg — APPROVED** [review]

**Review by ormergi — COMMENTED** [review]
I miss the removal of `CopyAlpineWithNonQEMUPermissions`, `DeleteAlpineWithNonQEMUPermissions`, and `RunPodInNamespace` from test/utils.go

**Pull request closed without merging**